### PR TITLE
:white_check_mark: Tested `starknet_getTransactionStatus`

### PR DESCRIPTION
**Pull request by Trantorian1**
:+1: Tests include:

- **invalid transaction**
- **transaction accepted on L1**
- **transaction accepted on L2**: uses latest transaction from latest block
- **transaction reverted on L1**

:-1: Test do not include:

- **Reverted L2 transactions**: complicated since this would require parsing the latest block for `REVERTED` transaction (which is not guaranteed). 
- **Received transaction**: transaction won't even be added to latest block until it is accepted, so this isn't possible unless a mock sequencer is set up to hold transaction in `RECEIVED` state.
- **Rejected**: again complicated, since `REJECTED` transactions are not added to blocks and cannot be found on explorers.

TLDR: a mock sequencer would be a good idea in the future.

### CODE REVIEW

**Comment by Trantorian1** [comment]
Formatted code to pass clippy checks in 64c4829d0cbd2466a69a3308d455b7165523ae8b
